src/dune_pkg/substs.mli

```diff
@@ -0,0 +1,25 @@
+open Stdune
+
+module Var : sig
+  type t =
+    { package : Dune_lang.Package_name.t option
+    ; variable : string
```

Comment:
Let's use proper types for this stuff. No strings in the public API please.

Reply:
Done. I couldn't think of anything better so I created a wrapper around `OpamVariable.t` with some conversion functions into `Dune_lang.t` and `Dyn.t`.